Model/Observer/ExtensionNotification.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Model\Observer;
+
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Magento\Framework\App\CacheInterface;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Notification\NotifierInterface;
+
+class ExtensionNotification implements ObserverInterface
+{
+    const CHECK_FREQUENCY = 604800; // one week
+
+    const REPOSITORY_URL = 'https://api.github.com/repos/algolia/algoliasearch-magento-2/releases/latest';
+
+    const CURL_OPT = [
+        'http' => [
+            'method' => 'GET',
+            'header' => [
+                'User-Agent: PHP',
+            ],
+        ],
+    ];
+
+    /** @var NotifierInterface */
+    private $notifier;
+
+    /** @var CacheInterface */
+    protected $cacheManager;
+
+    /** @var ConfigHelper */
+    private $configHelper;
+
+    private $repoData = null;
+
+    /**
+     * @param NotifierInterface $notifier
+     * @param CacheInterface $cacheManager
+     * @param ConfigHelper $configHelper
+     */
+    public function __construct(
+        NotifierInterface $notifier,
+        CacheInterface $cacheManager,
+        ConfigHelper $configHelper
+    ) {
+        $this->notifier = $notifier;
+        $this->cacheManager = $cacheManager;
+        $this->configHelper = $configHelper;
+    }
+
+    /**
+     * @param Observer $observer
+     *
+     */
+    public function execute(Observer $observer)
+    {
+        if ($this->getLastCheck() + self::CHECK_FREQUENCY > time()) {
+            return;
+        }
+
+        $this->checkExtensionVersion();
+        $this->setLastCheck();
+    }
+
+    /**
+     * Retrieve Last check time
+     *
+     * @return int
+     */
+    private function getLastCheck()
+    {
+        return $this->cacheManager->load('algoliasearch_notification_lastcheck');
+    }
+
+    /**
+     * Set last check time (now)
+     *
+     * @return $this
+     */
+    private function setLastCheck()
+    {
+        $this->cacheManager->save(time(), 'algoliasearch_notification_lastcheck');
+
+        return $this;
+    }
+
+    /**
+     * Retrieve Last checked version
+     *
+     * @return string
+     */
+    private function getLastCheckedVersion()
+    {
+        return $this->cacheManager->load('algoliasearch_notification_lastchecked_version');
+    }
+
+    /**
+     * Set last checked version
+     *
+     * @param $version
+     *
+     * @return $this
+     */
+    private function setLastCheckedVersion($version)
+    {
+        $this->cacheManager->save($version, 'algoliasearch_notification_lastchecked_version');
+
+        return $this;
+    }
+
+    private function checkExtensionVersion()
+    {
+        $versionFromRepository = $this->getLatestVersionFromRepository()->name;
```

Comment:
In case the call is not working (e.g. Github down, or customer has a dump/strict firewall), I believe it should fall back nicely (e.g. try/catch or something).

Reply:
I added a try/catch on the new model